### ISSUE TITLE
Zenkaku-space in doc comment is converted to \u{3000}

### ISSUE THREAD

**Issue by hatoo** [hatoo/cargo-snippet#15]
```
#[snippet]
/// zenkaku　space
fn foo() {}
```
cargo snippet

```
snippet foo
    /// zenkaku\u{3000}space
    fn foo() {}
```

This is not caused by #14.
I've confirmed that this issue on old cargo-snippet too.

**Comment by magurotuna** [comment]
I guess this is because proc_macro2 processes characters by `escape_debug`.

Related issue & PR:
- [Don't escape_default all non-ascii text by dtolnay · Pull Request #215 · alexcrichton/proc-macro2](https://github.com/alexcrichton/proc-macro2/pull/215/files)
- [Doc comments that include NON-ASCII characters not work as expected · Issue #772 · dtolnay/syn](https://github.com/dtolnay/syn/issues/772)

I tried some experiments in a playground [HERE](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=670b641b569a9099248b4b254cbd4601).
It gives us interesting facts that `\u{3001}` (= `、` non space character)  is escaped as we expect, but kinds of NON-ASCII space characters such as `\u{3000}`, `\u{2009}`  and `\u{2002}` are treated as EscapeUnicode.
In other words, `escape_debug` handles NON-ASCII spaces as special, I guess.

I'm going to continue to do more research on this.
